index.d.ts

```diff
@@ -0,0 +1,136 @@
+type KeyEffectDistance = {
+  distanceX: number;
+  distanceY: number;
+};
+
+interface KeyEffect {
+  initialX: number;
+  initialY: number;
+  name?: string;
+  description?: string;
+}
+
+interface KeyMap<E> {
+  keys: Record<string, E>;
+  name?: string;
+  description?: string;
+}
+
+interface KeyMappingConfig {
+  dpad?: KeyMap<KeyEffect & KeyEffectDistance>[];
+  tap?: KeyMap<KeyEffect>[];
+  swipe?: KeyMap<KeyEffect & KeyEffectDistance>[];
+}
+
+type DeviceRendererKeyMapping = {
+  enable(isEnable: boolean): void;
+  setConfig(config: KeyMappingConfig): void;
+  activeKeyMappingDebug(isTraceActivate?: boolean, isGridActivate?: boolean): void;
+};
+
+type VmEvent = 'beforeunload' | 'fingerprint' | 'gps' | 'BATTERY_LEVEL' | string // TODO Provide an exhaustive list
```

Comment:
TODO

Reply:
From what I've found, there are a few other events that are listed in the tests, but I don't know yet if the tests are exhaustive. Here are all values found in the tests:
```
ANDROID_ID
baseband
battery
BATTERY_STATUS
BLK
CLIPBOARD
diskio
fingerprint
framework
gps
IMEI
NETWORK
network_profile
settings
SYSTEM_PATCHER_LAST_RESULT
SYSTEM_PATCHER_STATUS
systempatcher
``` 
I'm trying to find a source of truth for these, either in the cloud player, maybe libcom repo, or something else.